cpp/models/abm/world.h

```diff
@@ -303,6 +303,133 @@ class World
      */
     void remove_testing_scheme(const LocationType& loc_type, const TestingScheme& scheme);
 
+    /**
+     * @brief Get a reference to a Person from this World.
+     * @param[in] id A person's PersonId.
+     * @return A reference to the Person.
+     * @{
+     */
+    Person& get_person(PersonId id)
+    {
+        assert(id.get() < m_persons.size());
+        return m_persons[id.get()];
+    }
+
+    const Person& get_person(PersonId id) const
+    {
+        assert(id.get() < m_persons.size());
+        return m_persons[id.get()];
+    }
+    /** @} */
+
+    /**
+     * @brief Get the number of Person%s of a particular #InfectionState for all Cell%s.
+     * @param[in] location A LocationId from the world.
+     * @param[in] t TimePoint of querry.
+     * @param[in] state #InfectionState of interest.
+     * @return Amount of Person%s of the #InfectionState in all Cell%s of the Location.
+     */
+    size_t get_subpopulation(LocationId location, TimePoint t, InfectionState state) const
+    {
+        return std::count_if(m_persons.begin(), m_persons.end(), [&](auto&& p) {
+            return p.get_location() == location && p.get_infection_state(t) == state;
+        });
+    }
+
+    /**
+     * @brief Get the total number of Person%s at the Location.
+     * @param[in] location A LocationId from the world.
+     * @return Number of Person%s in the location.
+     */
+    size_t get_number_persons(LocationId location) const
+    {
+        if (!m_is_local_population_cache_valid) {
+            build_compute_local_population_cache();
+        }
+        return m_local_population_cache[location.index];
```

Comment:
Please also add a test to cover `m_is_local_population_cache_valid`

Reply:
This check is there to cover the case where World::interact is called manually without a prior World::begin_step. The easy solution is to make interact private and remove this check.

And since this is purely internal state, we cannot test it, only cover it. And IMO that is a bad idea.